test/OrchardCore.Tests/Apis/ContentManagement/DeploymentPlans/ContentStepLuceneQueryTests.cs

```diff
@@ -33,6 +33,10 @@ public async Task ShouldUpdateLuceneIndexesOnImport()
 
             await context.PostRecipeAsync(recipe);
 
+            // Search indexes are no longer updated in a deferred task at the end of a shell scope
+            // but in a background job after the http request, so they are not already up to date.
+            await Task.Delay(2_000);
```

Comment:
These are risky ways to deal with waiting for the background job to finish. Such arbitrary delays make tests flaky, like you can see on the example of `LuceneQueryTests` in the most recent build too: [first attempt fails](https://github.com/OrchardCMS/OrchardCore/actions/runs/7468367647/attempts/1), [second attempt succeeds](https://github.com/OrchardCMS/OrchardCore/actions/runs/7468367647/attempts/2).

Please provide a way to wait for the results directly in these tests instead of relying on delays that might or might not work by coincidence.

Reply:
Sorry, I can't think of any other way to reproduce this problem at the moment, because the query: `RecentBlogPosts` is a `Lucene` query, and if it is looked up immediately after a push, the query results will be empty. The original code was to put the Lucene index in a synchronous thread, which would cause unnecessary waiting. If we were in a bulk import scenario, the thread creating the index would be too slow and the client would have to wait a long time for the index to complete

Reply:
You could perhaps poll the index in this test or somehow otherwise explicitly wait for indexing to finish?

Reply:
So... What are the conditions for stopping rotation training? With a  `StopWatch`?

Reply:
That would also be an implicit wait that may or may not work randomly, the same as `Task.Delay()`. Instead, add code that explicitly waits for the indexing to finish. Something like this from your other PR would be the minimum, a polling approach: https://github.com/OrchardCMS/OrchardCore/blob/eaf58d000017f65ed650804ac35867371bfb558e/src/OrchardCore/OrchardCore.Abstractions/BackgroundJobs/HttpBackgroundJob.cs#L38-L50 Or better, something awaitable.

Reply:
@Piedone  updated, although it seems strange. But I don't know if there's a better solution

Reply:
Thank you!